### PR TITLE
Sw 5027 setup header and footer

### PR DESCRIPTION
**Pull request by hiboabd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+#### Installing dependencies locally: 
+- `yarn install`
+- `go mod download`
+-------------------------------------------------------------------
+
+## Local development
+
+The application ran through Docker can be accessed on `localhost:8888/supervision/deputies/public-authority/`.
+
+**Note: Sirius is required to be running in order to authenticate. Ensure that after logging in, you redirect back to the correct port (`8888`)**
```
Comment (denialanderror) [inline README.md:17]: Is this true? I thought we were using json-server to mock authentication?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,50 @@
...
+
+## Local development
+
+The application ran through Docker can be accessed on `localhost:8888/supervision/deputies/public-authority/`.
+
+**Note: Sirius is required to be running in order to authenticate. Ensure that after logging in, you redirect back to the correct port (`8888`)** 
+
+To enable debugging and hot-reloading of Go files:
+
+`docker-compose -f docker/docker-compose.yml -f docker/docker-compose.dev.yml up --build`
```
Comment (denialanderror) [inline README.md:21]: this can just be `docker-compose -f docker/docker-compose.dev.yml up --build` now

File: cypress/plugins/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="cypress" />
```
Comment (denialanderror) [inline cypress/plugins/index.js:1]: Is this file needed? It doesn't appear to do anything

File: docker/puppeteer/pa11yci.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+    "defaults": {
+        "chromeLaunchConfig": {
+            "args": ["--no-sandbox"],
+            "ignoreHTTPSErrors": true
```
Comment (denialanderror) [inline docker/puppeteer/pa11yci.json:5]: Can we remove this now?

File: internal/server/deputy_hub.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+type deputyHubVars struct {
+	Path      string
+	XSRFToken string
+	Error     string
+	Errors    sirius.ValidationErrors
+}
+
+func loggingInfoForDeputyHub(client DeputyHubInformation, tmpl Template) Handler {
+	return func(perm sirius.PermissionSet, w http.ResponseWriter, r *http.Request) error {
+		fmt.Println("in handler")
```
Comment (denialanderror) [inline internal/server/deputy_hub.go:22]: remove this print statement

File: internal/server/deputy_hub.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+type deputyHubVars struct {
+	Path      string
+	XSRFToken string
+	Error     string
+	Errors    sirius.ValidationErrors
+}
+
+func loggingInfoForDeputyHub(client DeputyHubInformation, tmpl Template) Handler {
```
Comment (denialanderror) [inline internal/server/deputy_hub.go:20]: Not a fan of the naming of this function (I know this follows the pattern in Workflow) as it isn't logging anything, it is rendering a template.

File: internal/server/server.go [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+func New(logger Logger, client Client, templates map[string]*template.Template, prefix, siriusPublicURL, webDir string) http.Handler {
+	wrap := errorHandler(logger, client, templates["error.gotmpl"], prefix, siriusPublicURL)
+
+	mux := http.NewServeMux()
+	mux.Handle("/",
+		wrap(
+			loggingInfoForDeputyHub(client, templates["deputy-hub.gotmpl"])))
+
+	// mux.Handle("/health-check", healthCheck())
```
Comment (denialanderror) [inline internal/server/server.go:35]: Is this commented out because it is disabled? You should be able to follow the pattern in Workflow to enable it.

File: json-server/routes.json [hunk 8]
```diff
@@ -1,3 +1,3 @@
 {
-    "/api/v1/*": "/$1"  
+    "/api/v1/permissions": "/permissions"
```
Comment (denialanderror) [inline json-server/routes.json:2]: I don't think you need to make this change. The wildcards in the old version should have routed correctly.

File: web/template/deputy-hub.gotmpl [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{{ template "page" . }}
+
+{{ define "title" }} OPG Sirius Deputy Hub {{ end }}
+
+{{ define "main" }}
+
+    {{/* added to pass pa11y test- will be removed in later ticket */}}
```
Comment (denialanderror) [inline web/template/deputy-hub.gotmpl:7]: If you add TODO to the comment (i.e. `{{/* added to pass pa11y test- will be removed in later ticket */}}`), it is then obvious that this needs to be done in the future.

**Review by denialanderror — CHANGES_REQUESTED** [review]
Mostly minor comments but a few improvements.

File: internal/server/server.go (hunk 7 above)
Comment (hiboabd) [reply internal/server/server.go:35]: I didn't implement it because i didn't think it was part of the scope of this ticket - but it's a simple addition so i'll do that now

File: README.md (hunk 1 above)
Comment (hiboabd) [reply README.md:17]: Yeah that's correct

File: docker/puppeteer/pa11yci.json (hunk 4 above)
Comment (hiboabd) [reply docker/puppeteer/pa11yci.json:5]: Yep 🙌

**Review by denialanderror — APPROVED** [review]
